### PR TITLE
SALTO-3700 - Zendesk Adapter - Warn the user about bad format of trigger & automation actions

### PR DESCRIPTION
**Pull request by idoser96**
On deploy, if we see that the value type of trigger/automation notification_webhook is not an array, we will warn the user about it

---

None

---
_Release Notes_: 
Zendesk Adapter:
* In case a notification_webhook action of a trigger or an automation, is fetched in a bad format, the user will be warned on the deploy

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/zendesk-adapter/src/change_validators/bad_format_webhook_action.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    .map(getChangeData).filter(instance => POTENTIAL_BAD_FORMAT_TYPES.includes(instance.elemID.typeName))
+    .filter(instance => _.isArray(instance.value.actions))
+    .map((instance) : ChangeError | undefined => {
+      const webhookActions = instance.value.actions.filter(isAction).filter((action: Action) => action.field === 'notification_webhook')
+      if (webhookActions.some((action: Action) => !_.isArray(action.value))) {
+        const { typeName } = instance.elemID
+        return {
+          elemID: instance.elemID,
+          severity: 'Warning',
+          message: `${typeName} instance might not work properly`,
```
Comment (netama) [inline packages/zendesk-adapter/src/change_validators/bad_format_webhook_action.ts:53]: please use a more specific error message - maybe `... has an unexpected structure`? @tomermevorach wdyt?

**Review by netama — COMMENTED** [review]
LGTM with a few nits

File: packages/zendesk-adapter/src/change_validators/bad_format_webhook_action.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  isAdditionOrModificationChange,
+  isInstanceChange,
+} from '@salto-io/adapter-api'
+import { values as lowerDashValues } from '@salto-io/lowerdash'
+import joi from 'joi'
+import { AUTOMATION_TYPE_NAME, TRIGGER_TYPE_NAME } from '../constants'
+
+const { isDefined } = lowerDashValues
+
+type notificationWebhookAction = {
```
Comment (edenhassid) [inline packages/zendesk-adapter/src/change_validators/bad_format_webhook_action.ts:30]: nit:
```suggestion
type NotificationWebhookAction = {
```

**Review by edenhassid — APPROVED** [review]
looks great! added a few nits

File: packages/zendesk-adapter/src/change_validators/bad_format_webhook_action.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    value: unknown
+}
+
+const notificationWebhookAction = joi.object({
+  field: joi.string().valid('notification_webhook').required(),
+  value: joi.any().required(),
+}).unknown(true)
+
+const isNotificationWebhookAction = (value: unknown): value is notificationWebhookAction =>
+  notificationWebhookAction.validate(value).error === undefined
```
[the remark this answers is not in the document]
Comment (idoser96) [reply packages/zendesk-adapter/src/change_validators/bad_format_webhook_action.ts:41]: its ok if its not, doesn't have to be

**Pull request closed without merging**